packages/parameters/tests/helpers/resources.ts

```diff
@@ -349,7 +329,7 @@ class TestAppConfigWithProfiles extends Construct {
     this.profiles.forEach((profile) => {
       const appConfigConfigurationArn = Stack.of(fn).formatArn({
         service: 'appconfig',
-        resource: `application/${profile.applicationId}/environment/${profile.environmentId}/configuration/${profile.configurationProfileId}`,
+        resource: `application/${profile.application.applicationId!}/environment/${profile.deployTo![0].environmentId}/configuration/${profile.configurationProfileId}`,
```

Comment:
Is this non-null assertion needed?

I haven't checked in the IDE (can do tomorrow), but the [automated scanning is flagging it](https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=2524&id=aws-powertools_powertools-lambda-typescript&open=AY94zAtTdMQlEnSzB59V) as unnecessary.

If it's a false positive let me know and I'll mark it as such.

Reply:
I think I added it because my IDE was complaining. But I‘ll check it again.

Reply:
```suggestion
        resource: `application/${profile.application.applicationId}/environment/${profile.deployTo![0].environmentId}/configuration/${profile.configurationProfileId}`,
```

Reply:
Ok - I checked the code & the docs. The `applicationId` property is always defined and it's a string, so it's safe to remove the non-null assertion.

The other one on `deployTo` is valid, so we can keep it.